### ISSUE TITLE
Reduce sample/logging latency by using blocking

### ISSUE THREAD

**Issue by dwilliamson** [Celtoys/Remotery#19]
The message queue is polled every 10ms, in-between which the thread is put to sleep. Latency increases and messages have the potential to be discarded.

Add wakeup calls with blocking waits in the main thread (semaphores? events?) to process immediately.

**Comment by hlide** [comment]
There is an interesting blog post, [Semaphores are Surprisingly Versatile](http://preshing.com/20150316/semaphores-are-surprisingly-versatile). Especially the *AutoResetEvent* synchronization which is multi-platform and outperforms the original Windows *Auto-reset Event* because it calls a kernel function only if the current thread really needs to wait.

Using a sleep method is a very bad idea especially under Windows:

- You cannot wake up immediately as you can with a *Win32 SetEvent*.

- Usual minimal sleep time is around 25 ms even if you set a period below.

- You can lower this minimal sleep time with dedicated functions but they will affect the global system and provoke drawbacks because the scheduler will switch threads too many.

The thing I see is to call *signal* when enqueueing and to call *wait* when dequeueing. It should be fast enough since calling kernel function in *signal* is called if and only if at least one thread is already waiting.

**Comment by dwilliamson** [comment]
Yes, all this is known but it's very tricky to make all this cross-platform and integrated with Remotery's unique command queue that is MpSc while allocating all its variable-size message data from the queue itself.

Sleep is simple, ugly, predictable and far from ideal but seems to be working for most cases I've seen (I've used Remotery to profile UE4 with some 40 active threads with no issues).

If you have time to work on this yourself, that would be amazing. But right now it's not too much of a problem.
